### ISSUE TITLE
Inquiry about the Use of On-Demand API in simdjson

### ISSUE THREAD

**Issue by zhanglistar** [ClickHouse/ClickHouse#71720]
### Company or project name

Gluten project from BIGO.

### Question

I am currently exploring the integration of simdjson within ClickHouse and have noticed that the current implementation utilizes the DOM API. 

I would like to inquire whether there has been any consideration or discussion about adopting the on-demand API of simdjson instead. The on-demand API could potentially offer performance benefits and more efficient memory usage for certain use cases. Details can be found here: https://arxiv.org/abs/2312.17149.

I would appreciate any insights or thoughts on this matter. Thank you for your time!

**Comment by Algunenano** [comment]
https://github.com/ClickHouse/ClickHouse/issues/38621#issuecomment-1176347027 and following thread.

AFAIK nobody is currently looking into it. If somebody wants to explore it, that'd be a great experiment

**Comment by nickitat** [comment]
https://github.com/ClickHouse/ClickHouse/pull/39290

**Comment by alexey-milovidov** [comment]
Yes, we want to use the on-demand API for better performance, and any help will be very appreciated!
The work started in #39290, but now we need someone to take it and continue.

**Comment by zhanglistar** [comment]
Working on it.
